### PR TITLE
Support `min_delta` in early stopping.

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/dmlc/xgboost/issues/7136 .
Close https://github.com/dmlc/xgboost/pull/7066 .

### CODE REVIEW

**Comment by MarkGuryanov** [comment]
@trivialfis I'm not sure why we need to have both `abs_tol` and `min_delta`. Maybe I misunderstand the idea of `abs_tol`, but I thought of it as minimum change in the monitored metric in the direction of improvement to qualify as a substantial improvement. In this case we must not tolerate the event when metric gets worse and surely we must not consider this as an "improvement", because as I've described it in #7136, if we do this, the model will continue to train while metric on evaluation set gets slightly worse each step, and in the long run it will lead to overfitting, effectively nullifying the early stopping mechanism. Thus we must expect the metric to be strictly greater than previous best to consider it as an "improvement", 

It will be useful to rename parameter `abs_tol` to `min_delta` and fix the calculations (either by switching `+` to `-` or as it is done in #7066 - mathematically it is the same), but having both of them and requiring at least one to be 0 will be quite misleading for users IMHO.

Also, looking into APIs of early stopping callbacks from other frameworks (for example, [pytorch-lightning](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.callbacks.early_stopping.html#pytorch_lightning.callbacks.early_stopping.EarlyStopping) and [tensorflow](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/EarlyStopping#args)), I see only one parameter `min_delta` and no tolerance parameter.

**Comment by trivialfis** [comment]
My initial thought is just to allow some turbulence in validation results during training.  If that's not useful and doesn't make sense I can remove the `abs_tol` completely.

**Comment by trivialfis** [comment]
Your argument make sense to me.  Will follow up with the removal.

**Comment by trivialfis** [comment]
Removed.

**Review by MarkGuryanov — APPROVED** [review]
I think everything is fine now. Thank you!

**Review by RAMitchell — APPROVED** [review]
